bluetooth/src/iosMain/kotlin/scanner/Scanner.kt

```diff
@@ -122,39 +167,28 @@ actual class DefaultScanner internal constructor(
     }
 
     override suspend fun didStopScanning() {
-        centralManagers.forEach { centralManager ->
-            if (centralManager.state == CBCentralManagerStatePoweredOn) {
-                centralManager.stopScan()
-            }
-        }
-        discoveringDelegates.clear()
-        centralManagers.clear()
+        getCentralManager().stopScan()
     }
 
     override fun generateEnableSensorsActions(): List<EnableSensorAction> {
         // Trigger Enable Bluetooth popup
         return listOfNotNull(
             if (!bluetoothEnabledMonitor.isServiceEnabled) {
                 suspend {
-                    val options =
-                        mapOf<Any?, Any>(CBCentralManagerOptionShowPowerAlertKey to true)
-                    CBCentralManager(null, enabledQueue, options)
+                    getCentralManager()
                     bluetoothEnabledMonitor.isEnabled.first { it }
                 }
             } else null
         )
     }
 
     override suspend fun retrievePairedDevices(withServices: Set<UUID>) {
-        val awaitPoweredOn = EmptyCompletableDeferred()
-        val delegate = PoweredOnCBCentralManagerDelegate(this, awaitPoweredOn).freeze()
-        val centralManager = CBCentralManager(delegate, pairedDevicesQueue)
-        awaitPoweredOn.await()
+        val centralManager = getCentralManager()
```

Comment:
Can be lifted up into lazy property then (and no need to create it here on in scan)

Reply:
I don't think lazy variables can be suspend